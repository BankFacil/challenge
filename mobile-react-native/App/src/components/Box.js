import styled from '@emotion/native';
import { color, flexbox, layout, space, position, border, variant } from 'styled-system';

const Box = styled.View(
	layout,
	color,
	space,
	flexbox,
	position,
	border,
	variant({
		variants: {
			main: {
				backgroundColor: 'white',
				px: 2,
				flex: 1
			}
		}
	})
);

export default Box;

