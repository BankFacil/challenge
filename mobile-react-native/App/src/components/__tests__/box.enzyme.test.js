import React from 'react';
import { mount } from 'enzyme';
import Box from '../Box';

describe('<Box />', () => {
  it('should render component with spaces by theme', () => {
    const wrapper = mount(<Box m="3" p="2" />);
    console.log(wrapper.children().get(0).props.style);

    const valueMargin = wrapper.children().get(0).props.style.margin;
    const accertMargin = 16;

    const valuePadding = wrapper.children().get(0).props.style.padding;
    const accertPadding = 8;

    expect(valueMargin).toBe(accertMargin);
    expect(valuePadding).toBe(accertPadding);
  });
});
