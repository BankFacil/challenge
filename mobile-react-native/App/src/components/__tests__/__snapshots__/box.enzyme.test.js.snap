// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> can add a Todo with Enzyme 1`] = `
<emotion(View)
  m="3"
>
  <View
    m="3"
    style={
      Object {
        "margin": 16,
      }
    }
  >
    <View
      m="3"
      style={
        Object {
          "margin": 16,
        }
      }
    />
  </View>
</emotion(View)>
`;
